security/pkg/server/ca/node_auth.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ca
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	listerv1 "k8s.io/client-go/listers/core/v1"
+
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/kube/controllers"
+	"istio.io/istio/pkg/security"
+	"istio.io/istio/pkg/spiffe"
+)
+
+// NodeAuthorizer is a component that implements a subset of Kubernetes Node Authorization
+// (https://kubernetes.io/docs/reference/access-authn-authz/node/) for Istio CA. Specifically, it
+// validates that a node proxy which requests certificates for workloads on its own node is request
+// valid identities which run on that node (rather than arbitrary ones).
+type NodeAuthorizer struct {
+	podLister           listerv1.PodLister
+	podIndexer          *controllers.Index[*v1.Pod, SaNode]
+	trustedNodeAccounts map[types.NamespacedName]struct{}
+}
+
+const NodeSaIndex = "node+sa"
+
+func NewNodeAuthorizer(client kube.Client, trustedNodeAccounts map[types.NamespacedName]struct{}) (*NodeAuthorizer, error) {
```

Comment:
What does "trustedNodeAccounts" contains?

Reply:
It is a set of Trust Node Service Accounts - accounts that are valid to sue the new authorizor. In ambient the default is `kube-system/ztunnel`.

Reply:
I can see the service account part (NamespacedName), what is the struct it maps to?

Reply:
That is how sets are represented in Go

Reply:
I see. thanks.